cumulus/client/consensus/common/README.md

```diff
@@ -0,0 +1,27 @@
+<div align="center">
+
+<img
+alt="Polkadot logo" width="200"
+src="https://raw.githubusercontent.com/paritytech/polkadot-sdk/rzadp/readmes/docs/images/Polkadot_Logo_Horizontal_Pink_BlackOnWhite.png">
+
+# Cumulus-Specific Common Consensus Implementations
+
+This crate is part of the [Polkadot SDK](https://github.com/paritytech/polkadot-sdk/).
```

Comment:
How where those generated? Looks good IMHO, just asking.  
Eventually we should add a CI that ensures that they stay in sync with the rust files.

Reply:
> How where those generated? Looks good IMHO, just asking.

I did some one-off local bash scripting to generate the initial files, but then I relied on manual work to finish them, or to adapt already-existing readmes.

I guess it can be fully-automatic if:
- We accept the Cargo.toml's name as a title.
  - For example, crate `pallet-nft-fractionalization` can have an automatic title like `Pallet Nft Fractionalization`. But not a nicer `NFT Fractionalization Pallet` as I have handcrafted.
- The Cargo.toml's description is taken at face value, not rewritten into another sentence, etc.
Missing capitalization or a dot at the end can be auto-added.
- We leave a pack of handcrafted readmes untouched by the auto-generation script, as mentioned by @kianenigma

Reply:
> CI that ensures that they stay in sync with the rust files.

What do you mean exactly? In my mind, those readmes were never supposed to be out-of-sync with the rust files.

For example, this crate `cumulus-client-consensus-common` will always have a link to docs `(...)/cumulus_client_consensus_common`, and it will never contain anything other than "Cumulus-specific common consensus implementations" - at least I hope not :D

Reply:
I just wondered if you used something like this https://crates.io/crates/cargo-readme to generate the READMEs from the rust docs, since that can be put into CI to have up-to-date READMEs.  
Maybe i misunderstood what you are trying to do here?

Reply:
I was thinking the READMEs would be a small shell with a small intro, that invites the user to go to rendered rustdocs to consume the docs - not a copy of the rustdocs text converted to markdown.

Reply:
The main issue about generating README from .rs files is that they would not be able to parse `docify` stuff. 

Based on https://github.com/sam0x17/docify, there is a way to do this, but it would require first generating markdown files from .rs files, then tweaking the docify syntax, then running another process around it etc.

I recall talking with @sam0x17 last year about a CLI, but we never did it. 

In general, it is a cool idea, but not sure if it is a good bang for our buck atm. I would be happy with an auto-generated, mostly empty .md file that points to the right rustdoc page. It is less pretty, but works as good.